src/main/java/org/reaktivity/nukleus/ws/internal/WsController.java

```diff
@@ -96,7 +96,67 @@ public String name()
         return "ws";
     }
 
-    public CompletableFuture<Long> route(
+    public CompletableFuture<Long> routeInputNone(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            String protocol)
+    {
+        return route(Role.INPUT, State.NONE, source, sourceRef, target, targetRef, protocol);
+    }
+
+    public CompletableFuture<Long> routeInputNew(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            String protocol)
+    {
+        return route(Role.INPUT, State.NEW, source, sourceRef, target, targetRef, protocol);
+    }
+
+    public CompletableFuture<Long> routeInputEstablished(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            String protocol)
+    {
+        return route(Role.INPUT, State.ESTABLISHED, source, sourceRef, target, targetRef, protocol);
+    }
+
+    public CompletableFuture<Long> routeOutputNone(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            String protocol)
+    {
+        return route(Role.OUTPUT, State.NONE, source, sourceRef, target, targetRef, protocol);
+    }
+
+    public CompletableFuture<Long> routeOutputNew(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            String protocol)
+    {
+        return route(Role.OUTPUT, State.NEW, source, sourceRef, target, targetRef, protocol);
+    }
+
+    public CompletableFuture<Long> routeOutputEstablished(
+            String source,
+            long sourceRef,
+            String target,
+            long targetRef,
+            String protocol)
+    {
+        return route(Role.OUTPUT, State.ESTABLISHED, source, sourceRef, target, targetRef, protocol);
+    }
+
+    private CompletableFuture<Long> route(
```

Comment:
Please move now-private route method to end of file just before now-private unroute method.